### PR TITLE
fix: ensure “first one wins” on mountPointMap on identical devId; fil…

### PR DESCRIPTION
**Pull request by nocke**
Original bug encountered (Ubuntu-MATE 22.04)

> Error: ENOENT: no such file or directory, mkdir '/var/snap/firefox/common/host-hunspell/.Trash-1000'

What became clear after some investigation was, that all my files (under `/common/...` and `/depot/...` I do like work outside my home dir 😬) have a devId of 64768 (nothing wrong with that…), but that that ID is given twice  (!) in the list coming from procfs!

```
64768 ext4 / / ['rw', 'relatime'] /dev/mapper/nvme0n1p6_crypt ['rw', 'errors=remount-ro']
...(other entries)
64768 ext4 /usr/share/hunspell /var/snap/firefox/common/host-hunspell ['ro', 'noexec', 'noatime'] /dev/mapper/nvme0n1p6_crypt ['rw', 'errors=remount-ro']
...(other entries)

```

The first one looks sane (my “root mount“) the second doesn't. Reboot didn't help by the way.

I did two fixes (each one having effect in my case)
• first one wins (won't get „wiped“), so far last one won. – of course it's a gamble, if the more correct mount is truly listed earlier
• filter out 'snaps' (and while we're at it several others, like `/proc`, `/proc/`, `/proc/something…` but not  `/procedures` (perhaps that's a valid mount, eh?)

The result looks good on my machine (essentially like [`df`](https://linux.die.net/man/1/df), not missing anything. (nb: I do have LUKS encryption, distinct boot partition and a Samba-share, all good, all there)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "trash",
-	"version": "8.1.0",
+	"version": "8.1.1",
```
Comment (sindresorhus) [inline package.json:3]: Don't bump version number in pull requests.

**Comment by sindresorhus** [comment]
Make sure to run `npm test` locally before committing.

**Comment by nocke** [comment]
Hi.

• rolled back version
• fixed linting

File: lib/linux.js [hunk 2]
```diff
@@ -38,7 +38,17 @@ const trash = async (filePath, trashPaths) => {
 };
 
 export default async function linux(paths) {
-	const mountPointMap = new Map(procfs.processMountinfo().map(info => [info.devId, info.mountPoint]));
+	// • filter irrelevant
+	// • “first one wins” if id comes along twice
+	const mountPointMap = new Map();
+
+	for (const m of Object.values(procfs.processMountinfo())
```
Comment (sindresorhus) [inline lib/linux.js:45]: Don't use single-word variable name.

File: lib/linux.js [hunk 3]
```diff
@@ -38,7 +38,17 @@ const trash = async (filePath, trashPaths) => {
...
+		.filter(m => !/^\/(snap|run|sys|proc|dev)($|\/)/.test(m.mountPoint))) {
```
Comment (sindresorhus) [inline lib/linux.js:46]: Nitpick: Would be better to just add an if-statement in the loop for this and use `continue`.

File: lib/linux.js [hunk 4]
```diff
@@ -38,7 +38,17 @@ const trash = async (filePath, trashPaths) => {
 };
 
 export default async function linux(paths) {
-	const mountPointMap = new Map(procfs.processMountinfo().map(info => [info.devId, info.mountPoint]));
+	// • filter irrelevant
+	// • “first one wins” if id comes along twice
```
Comment (sindresorhus) [inline lib/linux.js:42]: The comment here is not clear. I think both lines could be a single code comment that explains it better.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: lib/linux.js (hunk 4 above)
Comment (nocke) [reply lib/linux.js:42]: I politely disagree on this one.

It's clearly 2 distinct aspects:
• filter irrelevant drives [line :46 does that, stuff that starts with `/snap`,`/run` etc...
• “first one wins” [the entire new loop :45-:50 does that, on drives that are not clearly „irrelevant“] but come along multiple times. Thus „first one wins“ as in „second one doesn't wipe out first“ (since only one can be considered for finding the right trash path…)

tl;dr: Keep it two lines, but I can make the explanation a bit longer. Just like that.

File: lib/linux.js (hunk 2 above)
Comment (nocke) [reply lib/linux.js:45]: Single-character-variables? (in extremly small scopes…), OK, can do :)

File: lib/linux.js (hunk 3 above)
Comment (nocke) [reply lib/linux.js]: Well, some people love chaining, some love if's... (usually I am in the if-camp, too)
Anyway, yes, changed to that.

And added `/var/snap` (as another and problem causing real-world encounter.)

**Comment by sindresorhus** [comment]
Thanks :)
